storage/block/block.go

```diff
@@ -335,32 +336,34 @@ func (dbb *databaseSeriesBlocks) MaxTime() time.Time {
 }
 
 func (dbb *databaseSeriesBlocks) BlockAt(t time.Time) (DatabaseBlock, bool) {
-	b, ok := dbb.elems[t]
+	b, ok := dbb.elems[xtime.ToUnixNano(t)]
 	return b, ok
 }
 
-func (dbb *databaseSeriesBlocks) AllBlocks() map[time.Time]DatabaseBlock {
+func (dbb *databaseSeriesBlocks) AllBlocks() map[xtime.UnixNano]DatabaseBlock {
 	return dbb.elems
 }
 
 func (dbb *databaseSeriesBlocks) RemoveBlockAt(t time.Time) {
-	if _, exists := dbb.elems[t]; !exists {
+	tNano := xtime.ToUnixNano(t)
+	if _, exists := dbb.elems[tNano]; !exists {
 		return
 	}
-	delete(dbb.elems, t)
+	delete(dbb.elems, tNano)
 	if !dbb.min.Equal(t) && !dbb.max.Equal(t) {
 		return
 	}
 	dbb.min, dbb.max = timeZero, timeZero
 	if len(dbb.elems) == 0 {
 		return
 	}
-	for k := range dbb.elems {
-		if dbb.min == timeZero || dbb.min.After(k) {
-			dbb.min = k
+	for key := range dbb.elems {
+		keyTime := key.ToTime()
+		if dbb.min == timeZero || dbb.min.After(keyTime) {
```

Comment:
How do you feel about exposing the time.Time methods in use  (`After`/`Before`/etc) on the `UnixNano` type? That way we could avoid the extra time.Time alloc

Reply:
talked offline, going to keep the scope to only map[time.Time] for now.